pandas/tests/series/test_missing.py

```diff
@@ -1315,16 +1328,17 @@ def test_spline_interpolation(self):
 
     @td.skip_if_no_scipy
     def test_spline_error(self):
-        # see gh-10633
+        # see GH-10633, GH-24014
         s = pd.Series(np.arange(10) ** 2)
         s[np.random.randint(0, 9, 3)] = np.nan
-        msg = "invalid method 'spline' to interpolate"
+        msg = "You must specify the order of the spline or polynomial"
         with pytest.raises(ValueError, match=msg):
             s.interpolate(method='spline')
```

Comment:
This pre-existing test is redundant with the `test_no_order` case above.  I can clean this up if reviewers would like.